packages/story-editor/src/app/rightClickMenu/useLayerSelect.js

```diff
@@ -0,0 +1,132 @@
+import createPolygon from '../canvas/utils/createPolygon';
+import { useCanvas } from '../canvas';
+import { getDefinitionForType } from '../../elements';
+
+function useLayerSelect({ menuItemProps, menuPosition, isMenuOpen }) {
+  const [isSubMenuOpen, setIsSubMenuOpen] = useState(false);
+  const { nodesById } = useCanvas(({ state: { nodesById } }) => ({
+    nodesById,
+  }));
+  const { currentPage, selectedElements, setSelectedElementsById } = useStory(
+    ({
+      state: { currentPage, selectedElements },
+      actions: { setSelectedElementsById },
+    }) => ({
+      currentPage,
+      selectedElements,
+      setSelectedElementsById,
+    })
+  );
+
+  useEffect(() => {
+    // Close submenu if the menu itself also closes.
+    if (!isMenuOpen) {
+      setIsSubMenuOpen(false);
+    }
+  }, [isMenuOpen]);
+
+  const { x, y } = menuPosition;
+  const getIntersectingElements = useCallback(() => {
+    if (!currentPage?.elements?.length || !Object.keys(nodesById)?.length) {
+      return [];
+    }
+    const clickedPoint = new SAT.Vector(x, y);
+    return currentPage.elements
+      .map((element) => {
+        const { id, rotationAngle } = element;
+        const node = nodesById[id];
+        if (!node) {
+          return null;
+        }
+        const { x: elX, y: elY, width, height } = node.getBoundingClientRect();
+        const elementP = createPolygon(rotationAngle, elX, elY, width, height);
+        return SAT.pointInPolygon(clickedPoint, elementP) ? element : null;
+      })
+      .filter((el) => el);
+  }, [currentPage, x, y, nodesById]);
+
+  const subMenuItems = useMemo(() => {
+    const intersectingElements = getIntersectingElements();
+    intersectingElements.reverse();
+    return intersectingElements.map((element) => {
+      const { id, isBackground, type } = element;
+      const { LayerContent } = getDefinitionForType(type);
+      return {
+        key: id,
+        supportsIcon: true,
+        icon: selectedElements[0].id === id ? <Icons.CheckmarkSmall /> : null,
+        label: isBackground ? (
+          <span>{__('Background', 'web-stories')}</span>
+        ) : (
+          <LayerContent element={element} />
+        ),
+        onClick: () => {
+          setSelectedElementsById({ elementIds: [id] });
+          trackEvent('context_menu_action', {
+            name: 'layer_selected',
+            element: type,
+            isBackground: isBackground,
+          });
+        },
+        ...menuItemProps,
+      };
+    });
+  }, [
+    getIntersectingElements,
+    menuItemProps,
+    setSelectedElementsById,
+    selectedElements,
+  ]);
+
+  // Only display if submenu has any items.
+  return subMenuItems.length > 0
+    ? {
+        label: __('Select Layer', 'web-stories'),
+        separator: 'bottom',
+        onClick: () => setIsSubMenuOpen(!isSubMenuOpen),
+        subMenuItems: isSubMenuOpen ? subMenuItems : [],
+        dismissOnClick: false,
+        SuffixIcon: Icons.ChevronRightSmall,
```

Comment:
This icon should change direction in RTL:
![Screen Shot 2022-01-19 at 12 09 29](https://user-images.githubusercontent.com/637548/150185774-7822fb23-0c06-4703-8b79-4539fd211392.png)